packages/twenty-server/src/core/open-api/utils/components.utils.ts

```diff
@@ -0,0 +1,81 @@
+import { ObjectMetadataEntity } from 'src/metadata/object-metadata/object-metadata.entity';
+import { FieldMetadataType } from 'src/metadata/field-metadata/field-metadata.entity';
+
+export const computeSchemaComponents = (
+  item: ObjectMetadataEntity,
+): {
+  type: string;
+  required?: string[];
+  properties: { [key: string]: { type: string; items: object } };
+} => {
+  const result = {
+    type: 'object',
+    properties: item.fields.reduce((node, field) => {
+      const itemProperties = {
+        type: 'string',
+      };
+
+      if (
```

Comment:
@martmull I think a `switch...case` is better for that
`if` condition one after another without using `else if` shouldn't happen here because you're modifying the same object in each condition and it must not overlap.